packages/web/hooks/limit-orders/use-place-limit.ts

```diff
@@ -258,19 +270,114 @@ export const usePlaceLimit = ({
   };
 };
 
-const useLimitPrice = () => {
+const useLimitPrice = ({
+  orderbookContractAddress,
+  quoteAssetDenom,
+  baseAssetDenom,
+  orderDirection,
+}: {
+  orderbookContractAddress: string;
+  quoteAssetDenom: string;
+  baseAssetDenom: string;
+  orderDirection: OrderDirection;
+}) => {
   // TODO: Fetch spot price from SQS
-  const spotPrice = useMemo(() => new Dec(1), []);
-  const [percentAdjusted, setPercentAdjusted] = useState(new Dec(0));
+  const { data: spotPrice, isLoading } =
+    api.edge.orderbooks.getSpotPrice.useQuery({
+      osmoAddress: orderbookContractAddress,
+      quoteAssetDenom,
+      baseAssetDenom,
+    });
+  const [orderPrice, setOrderPrice] = useState("");
+  const [manualPercentAdjusted, setManualPercentAdjusted] = useState("");
 
-  const adjustByPercentage = useCallback((percentage: Dec) => {
-    setPercentAdjusted(percentage);
-  }, []);
+  const adjustByPercentage = useCallback(
+    (percentage: Dec) => {
+      setOrderPrice(
+        formatPretty((spotPrice ?? new Dec(1)).mul(new Dec(1).add(percentage)))
```

Comment:
Hmm why do you want to default to 1 here?

Reply:
Can default to 0, might be safer